File: app/src/main/java/com/github/weatherapp/ui/adapter/MetricUnitAdapter.java
package com.github.weatherapp.ui.adapter;

import android.content.Context;
import android.support.annotation.Nullable;
import android.widget.ArrayAdapter;

public class MetricUnitAdapter extends ArrayAdapter<String> {
    private final String[] metricUnitValues;

    public MetricUnitAdapter(Context context, int resource, String[] metricUnitValues) {
        super(context, resource);
        this.metricUnitValues = metricUnitValues;

    }

    @Override
    public int getCount() {
        return metricUnitValues.length;
    }

    @Nullable
    @Override
    public String getItem(int position) {
        return metricUnitValues[position];
    }
}

File: app-kotlin/src/main/kotlin/com/github/weatherapp/ui/adapter/MetricUnitAdapter.kt
package com.github.weatherapp.ui.adapter

import android.content.Context
import android.widget.ArrayAdapter

class MetricUnitAdapter(context: Context, resource: Int, private val metricUnitValues: Array<String>) : ArrayAdapter<String>(context, resource) {

    override fun getCount(): Int {
        return metricUnitValues.size
    }

    override fun getItem(position: Int): String? {
        return metricUnitValues[position]
    }
}
